metagpt/roles/researcher.py

```diff
@@ -84,10 +84,17 @@ async def _react(self) -> Message:
         return msg
 
     def write_report(self, topic: str, content: str):
+        if not RESEARCH_PATH.exists():
+            RESEARCH_PATH.mkdir(parents=True)
         filepath = RESEARCH_PATH / f"{topic}.md"
         filepath.write_text(content)
 
 
 if __name__ == "__main__":
-    role = Researcher(language="en-us")
-    asyncio.run(role.run("dataiku vs. datarobot"))
+    import fire
+
+    async def main(topic: str, language="en-us"):
```

Comment:
set default topic valie

Reply:
Setting default values in the example is reasonable, but here, the command-line interface is provided through "fire," allowing users to directly execute python -m metagpt.roles.researcher "topic" in the terminal. If the user doesn't provide a "topic" argument, Fire will prompt them in the terminal.
```bash
python -m metagpt.roles.researcher
ERROR: The function received no value for the required argument: topic
Usage: researcher.py TOPIC <flags>
  optional flags:        --language

For detailed information on this command, run:
  researcher.py --help
```